pkg/plugins/repo/client_test.go

```diff
@@ -69,4 +69,19 @@ func Test_Download(t *testing.T) {
 		require.NoError(t, err)
 		require.Equal(t, 2, count, "should retry on error")
 	})
+
+	t.Run("it should do use gcom token when it's available", func(t *testing.T) {
+		expectedToken := "token-test"
+		fakeServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			token := r.Header.Get("Authorization")
+			require.Equal(t, "Bearer "+expectedToken, token, "gcom token should be set")
+			err := writeFakeZip(w)
+			require.NoError(t, err)
+		}))
+		defer fakeServer.Close()
+		cli := fakeServer.Client()
+		repo := Client{httpClient: *cli, httpClientNoTimeout: *cli, log: log.NewPrettyLogger("test"), grafanaComAPIToken: expectedToken}
+		_, err := repo.Download(context.Background(), fakeServer.URL, "", CompatOpts{})
+		require.NoError(t, err)
```

Comment:
Nit: Can you also make sure the HTTP handler is called?

```suggestion
		var gcomCalled bool
		fakeServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			token := r.Header.Get("Authorization")
			require.Equal(t, "Bearer "+expectedToken, token, "gcom token should be set")
			err := writeFakeZip(w)
			require.NoError(t, err)
			gcomCalled = true
		}))
		defer fakeServer.Close()
		cli := fakeServer.Client()
		repo := Client{httpClient: *cli, httpClientNoTimeout: *cli, log: log.NewPrettyLogger("test"), grafanaComAPIToken: expectedToken}
		_, err := repo.Download(context.Background(), fakeServer.URL, "", CompatOpts{})
		require.NoError(t, err)
		require.True(t, gcomCalled)
```